pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactedTopicImpl.java

```diff
@@ -304,8 +306,10 @@ static CompletableFuture<List<Entry>> readEntries(LedgerHandle lh, long from, lo
      * Getter for CompactedTopicContext.
      * @return CompactedTopicContext
      */
-    public Optional<CompactedTopicContext> getCompactedTopicContext() throws ExecutionException, InterruptedException {
-        return compactedTopicContext == null ? Optional.empty() : Optional.of(compactedTopicContext.get());
+    public Optional<CompactedTopicContext> getCompactedTopicContext() throws ExecutionException, InterruptedException,
```

Comment:
Please return a `CompletableFuture`.

DO NOT use `CompletableFuture#get`, which will block the thread.

Reply:
I added a new async method `getCompactedTopicContextAsync`, so there are two methods now
- `getCompactedTopicContext`
- `getCompactedTopicContextAsync`

There are `1` places that are calling the old method `getCompactedTopicContext`
- `topic internal stats`
- `HTTP requests: /metrics/`

I have changed the first use-case to `getCompactedTopicContextAsync` in the current PR, and the second one is hard to modify, we can separate a PR to improve the second case. And I added a timeout `30s` for the second case, which is better than calling it without a timeout.

Reply:
You can remove `getCompactedTopicContext` in the PR.

I checked the usages of `getCompactedTopicContext`, one is here, and the remaining calls are in the test package.

Reply:
see follows:
- https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/OpenTelemetryTopicStats.java#L472
- https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java#L328